charms/knative-operator/src/charm.py

```diff
@@ -1,104 +1,134 @@
 #!/usr/bin/env python3
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
-#
-# Learn more at: https://juju.is/docs/sdk
 
-"""Charm the service.
-
-Refer to the following post for a quick-start guide that will help you
-develop a new k8s charm using the Operator Framework:
-
-    https://discourse.charmhub.io/t/4208
-"""
+"""A Juju Charm for knative-operator."""
 
 import logging
+import traceback
 
+from jinja2 import Environment, FileSystemLoader
+from lightkube import ApiError, Client, codecs
 from ops.charm import CharmBase
-from ops.framework import StoredState
 from ops.main import main
-from ops.model import ActiveStatus
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus
+from ops.pebble import ChangeError, Layer
 
 logger = logging.getLogger(__name__)
 
 
-class OperatorTemplateCharm(CharmBase):
-    """Charm the service."""
-
-    _stored = StoredState()
+class KnativeOperatorCharm(CharmBase):
+    """A Juju Charm for knative-operator."""
 
     def __init__(self, *args):
         super().__init__(*args)
-        self.framework.observe(self.on.httpbin_pebble_ready, self._on_httpbin_pebble_ready)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
-        self.framework.observe(self.on.fortune_action, self._on_fortune_action)
-        self._stored.set_default(things=[])
-
-    def _on_httpbin_pebble_ready(self, event):
-        """Define and start a workload using the Pebble API.
-
-        TEMPLATE-TODO: change this example to suit your needs.
-        You'll need to specify the right entrypoint and environment
-        configuration for your specific workload. Tip: you can see the
-        standard entrypoint of an existing container using docker inspect
-
-        Learn more about Pebble layers at https://github.com/canonical/pebble
-        """
-        # Get a reference the container attribute on the PebbleReadyEvent
-        container = event.workload
-        # Define an initial Pebble layer configuration
-        pebble_layer = {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
+
+        self._app_name = self.model.app.name
+        self._namespace = self.model.name
+        self._operator_service = "/ko-app/operator"
+        self._container = self.unit.get_container(self._app_name)
+        self.env = Environment(loader=FileSystemLoader("src"))
+        self._resources_files = (
+            "auth_manifests.yaml",
+            "service_account.yaml",
+            "crds_manifests.yaml",
+            "istio_ns.yaml",
+            # Skipping config manifests for now
+            # "config_manifests.yaml",
```

Comment:
We were skipping this due to the `istio-system` namespace right? Given that we are creating this namespace for now, can this be uncommented?